examples/non-fungible/src/contract.rs

```diff
@@ -0,0 +1,386 @@
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    None,
+                    runtime.authenticated_signer(),
+                    source_account.owner,
+                )?;
+
+                let nft = self.get_nft(&token_id).await;
+                Self::check_account_authentication(
+                    None,
+                    runtime.authenticated_signer(),
+                    nft.owner,
+                )?;
+
+                Ok(self.transfer(nft, target_account).await)
+            }
+        }
+    }
+
+    async fn handle_application_call(
+        &mut self,
+        runtime: &mut ContractRuntime,
+        call: Self::ApplicationCall,
+        _forwarded_sessions: Vec<SessionId>,
+    ) -> Result<
+        ApplicationCallOutcome<Self::Message, Self::Response, Self::SessionState>,
+        Self::Error,
+    > {
+        match call {
+            Self::ApplicationCall::Mint { name, extra_data } => {
+                let execution_outcome = self
+                    .mint(
+                        AccountOwner::Application(runtime.authenticated_caller_id().unwrap()),
+                        name,
+                        extra_data,
+                    )
+                    .await;
+
+                Ok(ApplicationCallOutcome {
+                    execution_outcome,
+                    ..Default::default()
+                })
+            }
+
+            Self::ApplicationCall::Transfer {
+                source_owner,
+                token_id,
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    runtime.authenticated_caller_id(),
+                    runtime.authenticated_signer(),
+                    source_owner,
+                )?;
+
+                let nft = self.get_nft(&token_id).await;
+                Self::check_account_authentication(
+                    runtime.authenticated_caller_id(),
+                    runtime.authenticated_signer(),
+                    nft.owner,
+                )?;
+
+                let execution_outcome = self.transfer(nft, target_account).await;
+                Ok(ApplicationCallOutcome {
+                    execution_outcome,
+                    ..Default::default()
+                })
+            }
+
+            Self::ApplicationCall::Claim {
+                source_account,
+                token_id,
+                target_account,
+            } => {
+                Self::check_account_authentication(
+                    None,
+                    runtime.authenticated_signer(),
+                    source_account.owner,
+                )?;
+
+                let execution_outcome = if source_account.chain_id == system_api::current_chain_id()
+                {
+                    let nft = self.get_nft(&token_id).await;
+                    Self::check_account_authentication(
+                        None,
```

Comment:
Question: shouldn't this be `runtime.authenticated_caller_id()`?

Reply:
🤦🏻‍♂️ good catch!